idb/postgres/postgres.go

```diff
@@ -296,16 +296,31 @@ func (db *IndexerDb) AddBlock(block *bookkeeping.Block) error {
 				return fmt.Errorf("AddBlock() err: %w", err)
 			}
 		} else {
+			proto, ok := config.Consensus[block.BlockHeader.CurrentProtocol]
+			if !ok {
+				return fmt.Errorf(
+					"AddBlock() cannot find proto version %s", block.BlockHeader.CurrentProtocol)
+			}
+			protoChanged := !proto.EnableAssetCloseAmount
+			proto.EnableAssetCloseAmount = true
+
 			var wg sync.WaitGroup
 			defer wg.Wait()
 
-			// Write transaction participation in a parallel db transaction.
+			// Write transaction participation and possibly transactions in a parallel db
+			// transaction.
 			var err0 error
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
 
 				f := func(tx pgx.Tx) error {
+					if !protoChanged {
+						err := writer.AddTransactions(block, block.Payset, tx)
```

Comment:
I think you want want this in a separate `txWithRetry` so that it is written in parallel to `AddTransactionParticipation`?

I like this change but am not sure we need it right now. At the very least I think it's good to add a comment below near `AddTransactions`, something along the lines of:
> if proto.EnableAssetCloseAmount was already true, the transactions can be loaded with block.Payset instead of modifiedTransactions while the evaluator is running

Reply:
No, writing transactions and transaction participation together takes less time then eval + writing other state, so we don't need more parallelism; it would probably hurt.

I will add a comment.